packages/terra-pills/src/Pills.jsx

```diff
@@ -0,0 +1,383 @@
+  const isPillDeleted = useRef(false);
+
+  const generateRollUp = useCallback(() => {
+    const startIndex = PillsUtils.handleRollUp(pillsRef);
+    setUpdatedCount(startIndex);
+    setRollUpCount(React.Children.count(children) - startIndex);
+  }, [children]);
+
+  const generateExpansion = useCallback(() => {
+    setUpdatedCount(React.Children.count(children));
+  }, [children]);
+
+  useLayoutEffect(() => {
+    if (isSingleLine) {
+      generateRollUp();
+    } else {
+      generateExpansion();
+    }
+  }, [children, generateExpansion, generateRollUp, isSingleLine]);
+
+  const setTabIndex = (val) => {
+    const currentNode = currentPill.current ? pillsRef.current.querySelector(`[id=${currentPill.current}]`) : null;
+    if (currentNode) {
+      currentNode.setAttribute('tabIndex', val);
+    }
+  };
+
+  const focusCurrentNode = () => {
+    const currentNode = currentPill.current ? pillsRef.current.querySelector(`[id=${currentPill.current}]`) : null;
+    if (currentNode) {
+      currentNode.focus();
+    }
+  };
+
+  const focusPillsContainer = () => {
+    setContainerTabindex('0');
+    pillsRef.current.focus();
+  };
+
+  useEffect(() => {
+    const pills = [...pillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = pillsRef.current.querySelector('[data-terra-rollup-pill]');
+    const firstFocusableNode = PillsUtils.getFirstFocusableNode(pills, focusNode.current);
+
+    // if there is a roll Up pill, set tabindex to -1
+    if (rollUpPill) {
+      PillsUtils.setRollUpPillTabIndex(rollUpPill, '-1');
+    }
+
+    if (pills.length > 0) {
+      PillsUtils.setPillsTabIndex(pills, '-1');
+
+      // Determine the first focusable pill
+      if (firstFocusableNode !== -1) {
+        focusNode.current = firstFocusableNode;
+        currentPill.current = pills[focusNode.current].id;
+        setTabIndex('0');
+      } else if (isSingleLine && rollUpPill) {
+        currentPill.current = rollUpPill.getAttribute('id');
+        setTabIndex('0');
+      }
+    } else if (isSingleLine && rollUpPill && pills.length === 0) { // if the first pill is rollUp pill, set rollUp pill tabindex 0
+      currentPill.current = rollUpPill.getAttribute('id');
+      setTabIndex('0');
+      focusNode.current = 0;
+    }
+  }, [isSingleLine, updatedCount]);
+
+  useEffect(() => {
+    const pills = [...pillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = pillsRef.current.querySelector('[data-terra-rollup-pill]');
+    const nextFocusableNode = PillsUtils.getNextFocusableNode(pills, focusNode.current);
+    const prevFocusableNode = PillsUtils.getPreviousFocusableNode(pills, focusNode.current);
+    const startIndex = PillsUtils.handleRollUp(pillsRef);
+
+    if (isPillDeleted.current && React.Children.count(children) > 0) {
+      if (nextFocusableNode === -1 && prevFocusableNode === -1 && rollUpPill === null) {
+        focusPillsContainer();
```

Comment:
So are we still focusing the pills container  when all the pills have been removed ?